algorithms/src/snark/marlin/ahp/matrices.rs

```diff
@@ -213,11 +220,14 @@ pub(crate) fn arithmetize_matrix<F: PrimeField>(label: &str, matrix_evals: Matri
 
     end_timer!(matrix_time);
 
+    let mut labels: Vec<String> =
+        AHPForR1CS::<F, MarlinHidingMode>::index_polynomial_labels(&[label], vec![id]).collect_vec();
+
     MatrixArithmetization {
-        row: LabeledPolynomial::new("row_".to_string() + label, row, None, None),
-        col: LabeledPolynomial::new("col_".to_string() + label, col, None, None),
-        val: LabeledPolynomial::new("val_".to_string() + label, val, None, None),
-        row_col: LabeledPolynomial::new("row_col_".to_string() + label, row_col, None, None),
+        row: LabeledPolynomial::new(core::mem::take(&mut labels[0]), row, None, None),
+        col: LabeledPolynomial::new(core::mem::take(&mut labels[1]), col, None, None),
+        val: LabeledPolynomial::new(core::mem::take(&mut labels[2]), val, None, None),
+        row_col: LabeledPolynomial::new(core::mem::take(&mut labels[3]), row_col, None, None),
```

Comment:
Let's collect to a `VecDeque` and then use [pop_front](https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.pop_front)

Reply:
Good idea